src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -478,6 +478,8 @@ object OrderedRVD {
     // keys: RDD[kType]
     keys: RDD[RegionValue]): Array[OrderedRVPartitionInfo] = {
     val nPartitions = keys.getNumPartitions
+    if (nPartitions == 0)
```

Comment:
yeah. I think this method used to never expect an RDD with no partitions, but because shuffleAndAdjustBounds doesn't separately check for empty RDDs, using that instead of shuffle cause it to error in places where a hintpartitioner was specified and there were no partitions, I guess? I changed it here to handle the empty case because it seemed more robust that way.

Reply:
probably fixes this: https://github.com/hail-is/hail/issues/2876

Reply:
actually, can you add a test for that?

Reply:
fine if you want to break it up into 2 too

Reply:
ok, I'll add another test.

I think #2876 shouldn't be a problem any more, even without this? I took all the binarysearch stuff out during the partitioner changes. I can add a test for that issue, though.

Reply:
ah, ok. well we can close it either way :)

Reply:
let me test it first! :)